index.js

```diff
@@ -3696,15 +3718,25 @@ export const createCsharpBom = async (path, options) => {
   return {};
 };
 
-export const mergeDependencies = (dependencies, newDependencies) => {
+export const mergeDependencies = (
+  dependencies,
+  newDependencies,
+  parentComponent = {}
+) => {
   const deps_map = {};
+  const parentRef =
+    parentComponent && parentComponent["bom-ref"]
+      ? parentComponent["bom-ref"]
+      : undefined;
   const combinedDeps = dependencies.concat(newDependencies || []);
   for (const adep of combinedDeps) {
     if (!deps_map[adep.ref]) {
       deps_map[adep.ref] = new Set();
     }
     for (const eachDepends of adep["dependsOn"]) {
-      deps_map[adep.ref].add(eachDepends);
+      if (parentRef && eachDepends.toLowerCase() !== parentRef.toLowerCase()) {
```

Comment:
@cerrussell @ajmalab This is missing the case where parentRef is undefined causing the gradle bug. Can we improve this condition to handle !parentRef as well?